presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java

```diff
@@ -408,7 +415,7 @@ private synchronized PrioritizedSplitRunner pollNextSplitWorker()
         for (Iterator<TaskHandle> iterator = tasks.iterator(); iterator.hasNext(); ) {
             TaskHandle task = iterator.next();
             // skip tasks that are already running the configured max number of drivers
-            if (task.getRunningLeafSplits() >= maximumNumberOfDriversPerTask) {
+            if (task.getRunningLeafSplits() >= task.getMaxDriversPerTask().orElse(maximumNumberOfDriversPerTask)) {
```

Comment:
Use `max` here?

Reply:
I add a check when decoding the session property value to ensure that it's not greater than the system configured value.

Reply:
Without the session property, every query in the cluster will run with a per task driver count within [`guaranteedNumberOfDriversPerTask`, `maximumNumberOfDriversPerTask`]. We have use cases where we want to slow down certain type of query (e.g. to reduce traffic spike, or to reduce resource consumption per query), the new session property allow the user to be "nice" and lower the upper bound of drivers per task for one's query. And we also allow the user to lower the driver count to below the cluster wise minimum (i.e `maximumNumberOfDriversPerTask`). 

If the session property is set, the per task driver count of the query becomes [min(`guaranteedNumberOfDriversPerTask`, `sessionPropertyValue`), min(`maximumNumberOfDriversPerTask`, `sessionPropertyValue`)].

Reply:
Thanks for explanation